wpilib/wpilib/solenoidbase.py

```diff
@@ -23,13 +23,36 @@ def __init__(self, moduleNumber):
         """
         self.moduleNumber = moduleNumber
 
+    @staticmethod
+    def getAllStatic(moduleNumber):
+        """Read all 8 solenoids from the specified module as a
+        single byte.
+
+        :param moduleNumber: the module number to read.
+        :returns: The current value of all 8 solenoids on the module.
+        """
+        return hal.getAllSolenoids(moduleNumber)
+
     def getAll(self):
         """Read all 8 solenoids from the module used by this solenoid as a
         single byte.
 
         :returns: The current value of all 8 solenoids on this module.
         """
-        return hal.getAllSolenoids(self.moduleNumber)
+        return SolenoidBase.getAll(self.moduleNumber)
```

Comment:
This self-recursive call will never work.

Reply:
Whoops.

re "Static" - you answered yourself, its for backwards compatibility. For that particular naming convention, no reason. I'm not familiar enough with robotpy to know if a precedent has already been set for dealing with this particular situation. Figured I'd find out in the PR.